### PR TITLE
feat: source IaC results from LS, make progresses & scans cancelable [IDE-630]

### PR DESCRIPTION
**Pull request by bastiandoetsch**
### Description

- integrates IaC results from LS instead of CLI scan
- makes progresses cancelable
- cleans up redundant / old code

### Checklist

- [ ] Tests added and all succeed
- [x] Linted
- [x] CHANGELOG.md updated
- [ ] README.md updated, if user-facing

### Screenshots / GIFs

_Visuals that may help the reviewer. Please add screenshots for any UI change. GIFs are most welcome!_

### CODE REVIEW

File: src/main/kotlin/snyk/common/lsp/progress/ProgressManager.kt [hunk 1]
```diff
@@ -0,0 +1,235 @@
...
+                    indicator.cancel()
+                    progresses.remove(token)
+                }
+            }
+        }
+    }
+
+    private fun cancelProgress(token: String) {
+        try {
+            progresses[token]?.cancel()
```
Comment (ShawkyZ) [inline src/main/kotlin/snyk/common/lsp/progress/ProgressManager.kt:119]: am wondering if we need a to a mutex here

File: src/main/kotlin/io/snyk/plugin/Utils.kt [hunk 2]
```diff
@@ -215,11 +222,10 @@ fun controlExternalProcessWithProgressIndicator(
 
 fun isFileListenerEnabled(): Boolean = pluginSettings().fileListenerEnabled
 
-fun isSnykIaCLSEnabled(): Boolean = false
-
-
 fun isDocumentationHoverEnabled(): Boolean = Registry.get("snyk.isDocumentationHoverEnabled").asBoolean()
 
+fun isPreCommitCheckEnabled(): Boolean = Registry.get("snyk.issuesBlockCommit").asBoolean()
```
Comment (ShawkyZ) [inline src/main/kotlin/io/snyk/plugin/Utils.kt:227]: just to confirm this is disabled by default

File: src/main/resources/META-INF/plugin.xml [hunk 3]
```diff
@@ -42,6 +44,9 @@
     <registryKey key="snyk.timeout.results.waiting"
                  defaultValue="720000"
                  description="Snyk timeout (milliseconds) to wait for results during scan"/>
+    <registryKey key="snyk.issuesBlockCommit"
+                 defaultValue="false"
```
Comment (ShawkyZ) [inline src/main/resources/META-INF/plugin.xml:48]: ok nice I see it here

**Review by ShawkyZ — APPROVED** [review]
🚀🌑✨

File: src/main/kotlin/snyk/common/lsp/progress/ProgressManager.kt (hunk 1 above)
Comment (bastiandoetsch) [reply src/main/kotlin/snyk/common/lsp/progress/ProgressManager.kt:119]: progresses is a concurrentMap - so we don't

File: src/main/kotlin/io/snyk/plugin/ui/BranchChooserComboboxDialog.kt [hunk 4]
```diff
@@ -68,7 +68,6 @@ class BranchChooserComboBoxDialog(val project: Project) : DialogWrapper(true) {
         }
         runAsync {
             LanguageServerWrapper.getInstance().updateConfiguration()
-            LanguageServerWrapper.getInstance().sendScanCommand(project)
         }
     }
```
[the remark this answers is not in the document]
Comment (bastiandoetsch) [reply src/main/kotlin/io/snyk/plugin/ui/BranchChooserComboboxDialog.kt:72]: I think, updateConfiguration() already triggers a scan.

**Pull request merged**